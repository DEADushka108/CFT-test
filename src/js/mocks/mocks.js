export * from './const/const';
export * from './posts/posts';
export * from './store/store';
export * from './users/users';
export * from './comments/comments';
