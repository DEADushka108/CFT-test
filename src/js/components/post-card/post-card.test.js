import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';
import {Provider} from 'react-redux';
import {store, post, user} from '../../mocks/mocks';
import PostCard from './post-card.jsx';

it(`Should render PostCard component correctly`, () => {
  const tree = renderer
    .create((
      <Provider store={store}>
        <MemoryRouter>
          <PostCard
            post={post}
            userInfo={user}
          />
        </MemoryRouter>
      </Provider>
    ), {
      createNodeMock() {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
