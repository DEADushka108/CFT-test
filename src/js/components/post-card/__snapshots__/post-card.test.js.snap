// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render PostCard component correctly 1`] = `
<article
  className="post-card__notice"
>
  <div
    className="post-card__user"
  >
    <img
      alt="Bret"
      className="post-card__user-image"
      src="./img/content/no-user.png"
    />
    <a
      className="post-card__user-name"
      href="/user/1"
      onClick={[Function]}
    >
      Bret
    </a>
  </div>
  <div
    className="post-card__content"
  >
    <h3
      className="post-card__title"
    >
      sunt aut facere repellat provident occaecati excepturi optio reprehenderit
    </h3>
    <img
      className="post-card__image"
      src="./img/content/no-image.png"
    />
    <p
      className="post-card__text"
    >
      quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto
    </p>
  </div>
</article>
`;
